src/origin-matcher.js

```diff
@@ -1,4 +1,8 @@
 
+exports.generic = function (list) {
+  return list.length === 1 && list[0] === '*'
```

Comment:
I like the `allowAllOrigins` override. It seems like it would both simplify the implementation and make the user's intent explicit.